api/v1alpha1/openapi.yaml

```diff
@@ -2502,6 +2532,231 @@ components:
+        info:
+          type: string
+          description: 'Human readable information indicating details about last transition.'
+    DeviceRuntimeSummaryStatusType:
+      type: string
+      enum:
+      - 'Online'
+      - 'Degraded'
+      - 'Error'
+      - 'Rebooting'
+      - 'PoweredOff'
+      - 'Unknown'
+      x-enum-varnames:
+      - 'DeviceRuntimeOnline'
+      - 'DeviceRuntimeDegraded'
+      - 'DeviceRuntimeError'
+      - 'DeviceRuntimeRebooting'
+      - 'DeviceRuntimePoweredOff'
+      - 'DeviceRuntimeUnknown'
+    DeviceUpdateStatus:
+      type: object
+      properties:
+        status:
+          $ref: '#/components/schemas/DeviceUpdateSummaryStatus'
+          description: 'Status of the device update.'
+        phase:
+          $ref: '#/components/schemas/DeviceUpdatePhaseStatus'
+          description: 'The passphrase for the private key.'
+    DeviceUpdateSummaryStatus:
+      type: object
+      properties:
+        status:
+          $ref: '#/components/schemas/DeviceUpdateStatusType'
+          description: 'Status summary type of the device update.'
+        info:
+          type: string
+          description: 'Human readable information about the last update transition.'
+    DeviceUpdateStatusType:
+      type: string
+      enum:
+      - 'UpToDate'
+      - 'OutOfDate'
+      - 'Updating'
+      - 'Unknown'
+      x-enum-varnames:
+      - 'DeviceUpdateUpToDate'
+      - 'DeviceUpdateOutOfDate'
+      - 'DeviceUpdateUpdating'
+      - 'DeviceUpdateUnknown'
+    DeviceUpdatePhaseStatus:
+      type: object
+      properties:
+        status:
+          $ref: '#/components/schemas/DeviceUpdatePhaseType'
+          description: 'Status of the device update phase.'
+        info:
+          type: string
+          description: 'Human readable information about the last update phase transition.'
+    DeviceUpdatePhaseType:
+      type: string
+      enum:
+      - 'Downloading'
+      - 'Verifying'
+      - 'Updating'
+      - 'Rebooting'
+      - 'RollingBack'
+      - 'Failed'
+      x-enum-varnames:
+      - 'DeviceUpdatePhaseDownloading'
+      - 'DeviceUpdatePhaseVerifying'
+      - 'DeviceUpdatePhaseUpdating'
+      - 'DeviceUpdatePhaseRebooting'
+      - 'DeviceUpdatePhaseRollingBack'
+      - 'DeviceUpdatePhaseFailed'
+    DeviceIntegrityStatus:
+      type: object
+      properties:
+        status:
+          $ref: '#/components/schemas/DeviceIntegrityType'
+          description: 'Status of the device integrity.'
+        info:
+          type: string
+          description: 'Human readable information about the last integrity transition.'
+    DeviceIntegrityType:
+      type: string
+      enum:
+      - 'Passed'
+      - 'Failed'
+      - 'Unknown'
+      - 'Unsupported'
+      x-enum-varnames:
+      - 'DeviceIntegrityPassed'
+      - 'DeviceIntegrityFailed'
+      - 'DeviceIntegrityUnknown'
+      - 'DeviceIntegrityUnsupported'
+    DeviceResourceStatus:
+      type: object
+      properties:
```

Comment:
Do we need to have different schemas or can we have a list of generic resources (that a user may potentially extend in the future)?

Reply:
I went back and forth with that, generally speaking, API's should be explicit. Having a generalized list adds the complexity of validation is the enum element valid for the type. But I will update it to be generic.